src/viper/trans.ml

```diff
@@ -124,6 +125,17 @@ let self ctxt at =
   | Some id -> !!! at (LocalVar (!!! at id,!!! at RefT))
   | _ -> failwith "no self"
 
+let tr_string_literal ctxt at str =
+  let str_id =
+    match Hashtbl.find_opt ctxt.str_literal_to_fld str with
+    | None ->
+      let fresh_str_id = fresh_stamp "$text" in
```

Comment:
why not just `fresh_str_id = Hashtbl.length ctxt.str_literal_to_fld` ?

Reply:
Sounds reasonable, `Hashtbl.length` takes `O(1)` time while `fresh_stamp` looks for string definition in the map.

Reply:
Yeah, and first of all it's simpler